### ISSUE TITLE
Relations event?

### ISSUE THREAD

**Issue by zeus1921** [ProSavage/FactionsX#5]
hey there, i checked for the relations event but i didnt find it, is there anyway to have it?

**Comment by prosavage** [comment]
Hey there,

Yeah there is no 'relations' event, do you want a `FactionRelationChangeEvent` or something?

If you want to describe a implementation spec I can probably build this for you.

**Comment by zeus1921** [comment]
oh man that would be great!
my idea was to have an **alert discord channel** (with discordsrv) about faction relations, new factions, disbands, new players joining factions, so people could get all that info without having to be ingame.

Lets say someone wants to invade an faction, i would set **"hasBeenOfflineForSeconds":** to 2 hours or so, and whenever someone sets a faction to enemy, people would have time to prepare and defend their bases or whatsoever :D

So basically, having a general chat event about factions.

Thanks!

**Comment by prosavage** [comment]
Sure so it sounds like you have most of the stuff covered from other events.
I just need to fire an event for you each time a relation change happens

**Comment by prosavage** [comment]
@zeus1921 I went ahead and implemented a faction relation change event, it fires when a relation changes as requested.

I want you to go ahead and use the release attached in PR #6 and use the jar as a library to check if the event does what you want it to.

Make sure you run the release attached on your server as well.

**Comment by zeus1921** [comment]
there's premium addons in the zip, is that intended? anyway, i will test it tomorrow! big thanks!

**Comment by prosavage** [comment]
> there's premium addons in the zip, is that intended? anyway, i will test it tomorrow! big thanks!

yeah everything is free and open source now.

**Comment by zeus1921** [comment]
well now i need some help on the alerts.ylm thing

if you can help me
`  - Trigger: FactionRelationChangeEvent
    Channel: relalerts
    Conditions:
      - ??
    Embed:
      Color: "#ef6950"
      Author:
        ImageUrl: "{embedavatarurl}"
        Name: "FactionRelationChangeEvent message here"`

**Comment by zeus1921** [comment]
ok, so, all i got for now is this:

  - Trigger: FactionDisbandEvent
    Channel: relalerts
    Embed:
      Color: "#ef6950"
      Author:
        ImageUrl: "{embedavatarurl}"
        Name: "{username} disbanded the faction {faction}"

![image](https://user-images.githubusercontent.com/28808082/132376458-5b7df5a1-4116-4454-9d33-842ec9c97732.png)

But neither username or fplayer works :c

**Comment by zeus1921** [comment]
Update:

**${fPlayer.getName()} created the faction ${faction.tag}**
This is working
![image](https://user-images.githubusercontent.com/28808082/132382841-f5b32b06-f40a-4df6-9914-49c8f54c8daa.png)

But this isnt
${faction.tag}** changed relation with **${factionto.tag}** and they are now **${faction.getnewRelation()}

**Comment by prosavage** [comment]
okay so you are saying that new event is not working?
is it even firing for you at all?

**Comment by zeus1921** [comment]
the relation one no, not at all

This ones are working:
${fPlayer.getName()} have joined ${faction.tag}
${fPlayer.getName()} disbanded the faction ${faction.tag}
${fPlayer.getName()} created the faction ${faction.tag}

**Comment by prosavage** [comment]
I guess show me the config itself?
I thought you are using the actual API.

**Comment by zeus1921** [comment]
[here](https://pastebin.com/7zu97ntC)
also, can you give me the placeholders list?
